library/rtsp/src/main/java/com/google/android/exoplayer2/source/rtsp/RtspClient.java

```diff
@@ -161,11 +162,13 @@ public RtspClient(
       PlaybackEventListener playbackEventListener,
       String userAgent,
       Uri uri,
-      boolean debugLoggingEnabled) {
+      boolean debugLoggingEnabled,
+      @Nullable SocketFactory socketFactory) {
     this.sessionInfoListener = sessionInfoListener;
     this.playbackEventListener = playbackEventListener;
     this.userAgent = userAgent;
     this.debugLoggingEnabled = debugLoggingEnabled;
+    this.socketFactory = socketFactory;
```

Comment:
Please also add to javadoc

@param socketFactory The {@clink SocketFactory} for the client connection.

Also please add a missing

@param debugLoggingEnabled Whether to print RTSP messages

for us :)